Add I2C support for NXP frdm_ke17z board

ankeXiao opened the pull request:
The frdm_ke17z contains only one lpi2c0 instance and tested the `samples/drivers/i2c/target_eeprom` case.

ankeXiao committed:
boards: nxp: frdm_ke17z: enable i2c driver

Added lpi2c support for NXP FRDM-KE17Z board, which has only one
instance of lpi2c0.

Signed-off-by: Anke Xiao <anke.xiao@nxp.com>

ankeXiao commented:
Hi @hakehuang , please help to test this, thanks!

hakehuang commented:
> Hi @hakehuang , please help to test this, thanks!

fails on board testing
```
/home/ubuntu/nxp/frdm_ke17z/temp/frdm_ke17z_twister_test_only/twister-out/frdm_ke17z/samples/sensor/accel_polling/sample.sensor.accel_polling/handler.log
*** Booting Zephyr OS build v3.6.0-6332-g19970cb533da ***
sensor: device fxos8700@1c not ready.
```

hakehuang commented on boards/nxp/frdm_ke17z/Kconfig.defconfig:
```diff
@@ -0,0 +1,12 @@
+if BOARD_FRDM_KE17Z
+
+config FXOS8700_DRDY_INT1
+	default y
```
this is DNP in revB board, so we should not enable it

ankeXiao replied:
Thanks for checking and testing, I will drop the sensor related configuration

ankeXiao committed:
samples: drivers: i2c: target_eeprom: add frdm_ke17z support

Added i2c configuration to test target_eeprom case for frdm_ke17z
Added `sample.drivers.i2c.target.kinetis` in sample.yaml for frdm_ke17z
and frdm_ke17z512 automatically run the tests

Signed-off-by: Anke Xiao <anke.xiao@nxp.com>

hakehuang commented on samples/drivers/i2c/target_eeprom/sample.yaml:
```diff
@@ -6,4 +6,6 @@ tests:
       - lpcxpresso55s69/lpc55s69/cpu0
+      - frdm_ke17z
+      - frdm_ke17z512
     harness: TBD
```
if you enable it  please change harness like below
```
   harness: console
    harness_config:
      type: multi_line
      regex:
        - ".*: Cipher Sample"
        - ".*: ECB mode ENCRYPT - Match"
        - ".*: ECB mode DECRYPT - Match"
        - ".*: CBC mode ENCRYPT - Match"
        - ".*: CBC mode DECRYPT - Match"
        - ".*: CTR mode ENCRYPT - Match"
        - ".*: CTR mode DECRYPT - Match"
```

ankeXiao replied:
Thanks for reminding, I have updated the sample.yaml to support them

ankeXiao commented:
Hi All, please help to review this PR again, tested `samples/drivers/i2c/target_eeprom` to replace the fxos8700 sensor sample as it is DNP on RevB.

hakehuang commented:
board testing PASS.

```
2024-07-05 16:19:13,401 - twister - DEBUG - DEVICE: *** Booting Zephyr OS build v3.6.0-6332-g3b55f4face57 ***
2024-07-05 16:19:13,402 - twister - DEBUG - DEVICE: i2c target sample
2024-07-05 16:19:13,402 - twister - DEBUG - HARNESS:Console:EXPECTED(1/3):'i2c target sample'
2024-07-05 16:19:13,405 - twister - DEBUG - DEVICE: i2c target driver registered
2024-07-05 16:19:13,405 - twister - DEBUG - HARNESS:Console:EXPECTED(2/3):'i2c target driver registered'
2024-07-05 16:19:14,407 - twister - DEBUG - DEVICE: i2c target driver unregistered
2024-07-05 16:19:14,407 - twister - DEBUG - HARNESS:Console:EXPECTED(3/3):'i2c target driver unregistered'
2024-07-05 16:19:14,408 - twister - DEBUG - run status: frdm_ke17z/samples/drivers/i2c/target_eeprom/sample.drivers.i2c.target.kinetis passed
```
